src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_CAPACITY = 16;
+    private Node<K, V> [] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        threshold = (int) LOAD_FACTOR * table.length;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int index = hash(key);
+        Node<K,V> newNode = new Node<>(key, value, null);
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+        } else {
+            Node<K, V> current = table[index];
+            while (current != null) {
+                if (equalsKey(current.key, key)) {
+                    current.value = value;
+                    break;
+                }
+                if (current.next == null) {
+                    current.next = newNode;
+                    size++;
+                    break;
+                }
+                current = current.next;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (table[hash(key)] == null) {
+            return null;
+        } else if (table[hash(key)].next != null) {
+            Node<K, V> current = table[hash(key)];
+            while (current != null) {
+                if (equalsKey(current.key, key)) {
+                    return current.value;
+                }
+                current = current.next;
+            }
+            return null;
+        }
+        return table[hash(key)].value;
```

Comment:
```suggestion
       Node<K, V> current = table[hash(key)];
            while (current != null) {
                if (equalsKey(current.key, key)) {
                    return current.value;
                }
                current = current.next;
            }
            return null;
```
це можна спростити таким чином